### PR TITLE
CircleCI workflow triggers: handle integration branches

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -389,24 +391,24 @@ workflows:
...
+          cron: "20 17 * * 0,3"
           filters:
             branches:
               only:
                 - master
+                - /^feature-.*$/
     jobs:
-      - performance_ci_automation:
-          context: common
+      - build-macos
```
Comment (danni-m) [inline .circleci/config.yml:413]: @rafie what about `performance_ci_automation`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -345,20 +345,20 @@ workflows:
   build_and_package:
     jobs:
       - lint:
-          <<: *on-any-branch
+          <<: *not-on-integ-branch
```
Comment (danni-m) [inline .circleci/config.yml:348]: @rafie why cancel it on integration branches?

Comment (rafie) [reply .circleci/config.yml:348]: @danni-m We'll always have this on PR (non-integ) branches, so it's basically an annoyance (preventing one from merging a PR) or a mere duplication. If by an obscure way someone has managed to bypass lint on a PR, the next PR will catch it. WDYT?

File: .circleci/config.yml (hunk 1 above)
Comment (rafie) [reply .circleci/config.yml:413]: @danni-m `performance_ci_automation` just runs under `nightly`. So there's `nightly` and `nightly-twice-a-week`.

File: .circleci/config.yml (hunk 2 above)
Comment (danni-m) [reply .circleci/config.yml:348]: @rafie OK, i see.

**Review by danni-m — APPROVED** [review]

**Pull request merged**